prime-router/src/main/kotlin/fhirengine/translation/HL7toFhirTranslator.kt

```diff
@@ -56,7 +37,18 @@ class HL7toFhirTranslator internal constructor(
         hl7Message: Message,
     ): HL7MessageModel {
         val messageTemplateType = getMessageTemplateType(hl7Message)
-        return defaultMessageTemplates[messageTemplateType]
+
+        // if the requested message templates have been previously loaded, return the stored templates.
+        // otherwise, load the templates
+        val messageTemplate = if (messageTemplates.containsKey(configFolderPath)) {
```

Comment:
Templates are stored for performance reasons; otherwise integration test runtime is greatly extended while new instances of ResourceReader load the templates over and over again